pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -138,6 +140,10 @@ func (cl *Client) RegisterEventHandler(kind config.GroupVersionKind, handler mod
 
 // Run the queue and all informers. Callers should  wait for HasSynced() before depending on results.
 func (cl *Client) Run(stop <-chan struct{}) {
+	if cl.started {
```

Comment:
I don't get it. How are we calling Run twice? we call `incluster.NewController().Run()`, not reuse the same

Reply:
We may run it again in IstiodAnalyzer.Init `	go store.Run(cancel)`

what do you mean by `not reuse the same`

Reply:
Hi @howardjohn, I believe the same `s.RWConfigStore` containing `crdclient.Client` is used every time `incluster.NewController().Run()` runs, and `s.RWConfigStore` was only initialized once at the very beginning. 

Below are debugging lines I added to troubleshoot this issue and the actual log output. Hope this can show it better. From the log, you can see the same pointer `0xc001390000` of `Client` appears every time with different stop channels and the queue with id `m6g2psn2nl` has been closed later.

```go
func (cl *Client) Run(stop <-chan struct{}) {
	t0 := time.Now()
	log.Debug(string(debug.Stack()))
	cl.logger.Infof("--------------------------debug Starting Pilot K8S CRD controller %p, %#v, %p", cl, cl.queue, stop)
```

```log
2023-10-13T05:57:48.428132Z     info    kube    --------------------------debug Starting Pilot K8S CRD controller 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:false, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc000a944e0 controller=crd-controller
2023-10-13T05:58:01.868086Z     info    kube    --------------------------debug Starting Pilot K8S CRD controller 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:false, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc0041320c0 controller=crd-controller
2023-10-13T08:10:55.898751Z     info    kube    controller terminated 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc000a944e0      controller=crd-controller
2023-10-13T08:12:29.658852Z     info    kube    controller terminated 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc0041320c0      controller=crd-controller
2023-10-13T08:13:39.929971Z     info    kube    --------------------------debug Starting Pilot K8S CRD controller 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc005784060  controller=crd-controller
2023-10-13T08:13:39.930455Z     info    kube    controller terminated 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc005784060      controller=crd-controller
2023-10-13T08:14:07.900487Z     info    kube    --------------------------debug Starting Pilot K8S CRD controller 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc009b4c0c0  controller=crd-controller
2023-10-13T08:14:07.900979Z     info    kube    controller terminated 0xc001390000, &queue.queueImpl{delay:1000000000, tasks:[]queue.Task{}, cond:(*sync.Cond)(0xc0012ce3c0), closing:true, closed:(chan struct {})(0xc0012f6360), closeOnce:(*sync.Once)(0xc0012a2fc0), id:"m6g2psn2nl"}, 0xc009b4c0c0      controller=crd-controller
```